packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -323,12 +322,56 @@ export function buildExtraViteConfig(
 		// knownJsSrcExtensions: options.extensions
 	};
 
-	extraViteConfig.optimizeDeps = buildOptimizeDepsForSvelte(
-		svelteDeps,
-		options,
-		config.optimizeDeps
-	);
+	// extra handling for svelte dependencies in the project
+	const depsConfig = await crawlFrameworkPkgs({
+		root: options.root,
+		isBuild: options.isBuild,
+		viteUserConfig: config,
+		isFrameworkPkgByJson(pkgJson) {
+			return !!pkgJson.svelte;
+		},
+		isSemiFrameworkPkgByJson(pkgJson) {
+			return !!pkgJson.dependencies?.svelte || !!pkgJson.peerDependencies?.svelte;
+		},
+		isFrameworkPkgByName(pkgName) {
+			const isNotSveltePackage = isCommonDepWithoutSvelteField(pkgName);
+			if (isNotSveltePackage) {
+				return false;
+			} else {
+				return undefined;
+			}
+		}
+	});
+
+	log.debug('crawl svelte packages result', depsConfig);
 
+	if (options.prebundleSvelteLibraries) {
+		depsConfig.optimizeDeps = { include: [], exclude: [] };
+	} else if (options.disableDependencyReinclusion === true) {
+		depsConfig.optimizeDeps.include = depsConfig.optimizeDeps.include.filter(
+			(dep) => !dep.includes('>')
+		);
+	} else if (Array.isArray(options.disableDependencyReinclusion)) {
+		const disabledDeps = options.disableDependencyReinclusion;
+		const isDisabled = (dep: string) => {
+			// `crawlFrameworkPkgs` always return `{dep} > {something}` format for deep includes
+			return disabledDeps.some((disabledDep) => dep.includes(`${disabledDep} >`));
+		};
+		depsConfig.optimizeDeps.include = depsConfig.optimizeDeps.include.filter(
+			(dep) => !isDisabled(dep)
+		);
+	}
```

Comment:
Currently doing this very cheeky handling for `disableDependencyReinclusion` which should still preserve it's original function.